Undefined symbols for architecture x86_64

### Steps to Reproduce
Build a project with wasmtime on Macos x86_64 

### Expected Results

What do you expect to happen?
Successfull build

### Actual Results

```rust
Undefined symbols for architecture x86_64:
            "wasmtime::runtime::vm::stack_switching::stack::unix::x86_64::wasmtime_continuation_start::hdc1ced40823a7b01", referenced from:
                wasmtime::runtime::vm::libcalls::cont_new::hd74388a76ef1e7b1 in libwasmtime-866a24c229d0a23b.rlib[18](wasmtime-866a24c229d0a23b.wasmtime.326c71a07024d4d1-cgu.15.rcgu.o)
          ld: symbol(s) not found for architecture x86_64
          clang: error: linker command failed with exit code 1 (use -v to see invocation) 
```

### Versions and Environment

Wasmtime version or commit:
40.0.0

Operating system:
Macos 
Architecture:
x86_64 

### Extra Info
Work on Windows and Ubuntu hosts 

Anything else you'd like to add?
I get this on GHA runners. 
With 39.0.0, get no issues

--- a/crates/wasmtime/src/runtime/vm/stack_switching/stack/unix.rs
+++ b/crates/wasmtime/src/runtime/vm/stack_switching/stack/unix.rs
@@ -258,7 +258,7 @@ impl VMContinuationStack {
 
             let to_store = [
                 // Data near top of stack:
-                (0x08, (wasmtime_continuation_start as *const ()).addr()),
+                (0x08, wasmtime_continuation_start_address().addr()),
                 (0x10, tos.sub(0x10).addr()),
                 (0x18, tos.sub(0x40 + args_data_size).addr()),
                 (0x20, usize::try_from(args_capacity).unwrap()),

--- a/crates/wasmtime/src/runtime/vm/stack_switching/stack/unix/x86_64.rs
+++ b/crates/wasmtime/src/runtime/vm/stack_switching/stack/unix/x86_64.rs
@@ -7,6 +7,11 @@
 
 use core::arch::naked_asm;
 
+#[inline(never)] // FIXME(rust-lang/rust#148307)
+pub fn wasmtime_continuation_start_address() -> *const () {
+    wasmtime_continuation_start as *const ()
+}
+
 // This is a pretty special function that has no real signature. Its use is to
 // be the "base" function of all fibers. This entrypoint is used in
 // `wasmtime_continuation_init` to bootstrap the execution of a new fiber.

